Source/WebCore/loader/SubstituteData.h

```diff
@@ -51,7 +51,7 @@ class SubstituteData {
     {
     }
 
-    bool isValid() const { return m_content != nullptr; }
+    bool isValid() const { return m_content != nullptr || m_response.isRedirection(); }
```

Comment:
This seems suspicious.  Why would a redirection response but no content be a valid substitute data?